lib/gruf/server.rb

```diff
@@ -90,9 +90,8 @@ def start!
         logger.info { "Starting gruf server at #{@hostname}..." }
         server.run_till_terminated_or_interrupted(KILL_SIGNALS)
       end
-      server.wait_till_running
```

Comment:
What happens if the server fails to bind on a socket or similar? Will it raise an exception and prevent us having a hung server?

Reply:
The thread should propagate the error up, as per GRPC, which maintains a mutex and synchronized state on the underlying channel and server. (They're essentially now doing in gRPC core what gruf was doing prior to 2.13.0.) The `wait_till_running` is unnecessary - we were only using it to try and maintain 100% accuracy on the process title (see the PR description for more details).